File: sort/insertSort.py
#! python3
# insertSort.py-插入排序

def insertSort(arr):
	for i in range(1, len(arr)):
		temp = arr[i]							# 摸下一张牌
		j = i
		while (j > 0 and arr[j-1] > temp):		
			arr[j] = arr[j-1]				# 向左移动空位
			j -= 1
		arr[j] = temp							# 放入牌
	return arr

# 测试
arr = [-2, 21, 3, 23, 10, -3, 0, -1]
arr = insertSort(arr)
print(arr)

File: data_structure_js/insertSort.js
// 插入排序

function insertSort(arr) {
	for (var i = 1; i < arr.length; i++) {
		var temp = arr[i];
		var j = 0;
		for (j = i; j > 0 && arr[j - 1] > temp; j--) {
			arr[j] = arr[j - 1];
		}
		arr[j] = temp;
	}
}

// 测试
var N = 4;
var arr = [];
for (var i = 0; i < N; i++) {
	arr.push(Math.floor(Math.random() * N * 2));
}
console.log(arr);
insertSort(arr);
console.log(arr);
